server/instances.go

```diff
@@ -145,53 +145,19 @@ func (p *Plugin) InstallInstance(newInstance Instance) error {
 	var updated *Instances
 	err := UpdateInstances(p.instanceStore,
 		func(instances *Instances) error {
-			var previousInstance Instance
-			var err error
-			if instances != nil && len(instances.IDs()) > 0 && instances.checkIfExists(newInstance.GetID()) {
-				previousInstance, err = p.instanceStore.LoadInstance(newInstance.GetID())
-				if err != nil {
-					p.API.LogError("Error occurred while fetching the instance", "ID", newInstance.GetID(), "Error", err.Error())
-					return err
-				}
+			if instances == nil {
+				return errors.New("received nil 'instances' in UpdateInstances callback")
 			}
 
-			if !p.enterpriseChecker.HasEnterpriseFeatures() {
-				if instances != nil && len(instances.IDs()) > 0 {
-					// Override the enterprise check if a user is migrating from JWT to OAuth
-					if !(instances.checkIfExists(newInstance.GetID()) && previousInstance.Common().Type == CloudInstanceType && newInstance.Common().Type == CloudOAuthInstanceType) {
-						return errors.Errorf(licenseErrorString)
-					}
-				}
+			if !p.enterpriseChecker.HasEnterpriseFeatures() && len(instances.IDs()) > 0 && !instances.checkIfExists(newInstance.GetID()) {
```

Comment:
We are allowing to install multiple instances over each other with the same instance ID. Do we need to support this?

Reply:
@raghavaggarwal2308 I decided to go with this to minimize risk on the changes here, since this is how it was already working before this PR. I went through the install flow a few times with `/jira setup` and `/jira instance install cloud-oauth` `/jira instance uninstall cloud-oaith` several times in different orders to try and generate some issue, and it seemed to all work flawlessly in every case.

I think having it how it currently works is fine. Installing over the same instance effectively just kicks off the setup flow for that instance type.